samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -153,6 +153,10 @@ public int getAccessLogSamplingRatio(String storeName) {
   }
 
   public Optional<String> getStorageFactoryClassName(String storeName) {
+    //TODO HIGH dchen remove this after changing restore factory to factories.
```

Comment:
I don't think this will fix it, you would likely have to change `getStoreNames` so it does not return the store restore/backend factory config as the store name.